### PR TITLE
Create the temp folder for 'tools' restoration in the project directory.

### PR DESCRIPTION
**Pull request by Sridhar-MS**
This is a fix for regression caused by #877. It is required to create the
temp folder in the project directory so that same nuget.config resolution
happens for both 'dependencies' and 'tools'. See #844.

### CODE REVIEW

**Comment by brthor** [comment]
:shipit:

**Comment by Sridhar-MS** [comment]
@dotnet-bot test OSX Debug Build please.

**Comment by Sridhar-MS** [comment]
@anurse Do you think this a nuget xplat issue?
http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/release_osx_prtest/748/console

**Comment by Sridhar-MS** [comment]
@anurse Ignore the last comment. For stage0 we still use dnx.

**Comment by Sridhar-MS** [comment]
@dotnet-bot test OSX Release Build please.

**Comment by anurse** [comment]
I've seen that before and I think I fixed it with one of the script changes in my NuGet PR. Try rebasing on `master`

File: src/dotnet-restore/Program.cs [hunk 1]
```diff
@@ -124,7 +124,7 @@ private static void RestoreTools(Project project, RestoreTask restoreTask, bool
 
         private static void RestoreTool(LibraryRange tooldep, RestoreTask restoreTask, bool quiet)
         {
-            var tempRoot = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var tempRoot = Path.Combine(restoreTask.ProjectDirectory, Guid.NewGuid().ToString());
```
Comment (davidfowl) [inline src/dotnet-restore/Program.cs]: Put it in the bin or obj folder.

Comment (Sridhar-MS) [inline src/dotnet-restore/Program.cs]: At line [130](https://github.com/dotnet/cli/pull/898/files#diff-67e89a17951b6cc7bb7ddaee1f4620a9R130) it is indeed get put in the bin folder.

**Comment by Sridhar-MS** [comment]
@dotnet-bot test this please

**Comment by piotrpMSFT** [comment]
@Sridhar-MS if this is intended to go into 1.0, please retarget the PR to rel/1.0.0.

**Comment by Sridhar-MS** [comment]
Already merged into rel/1.0.0.

**Pull request closed without merging**